### PR TITLE
DOC: Additions/updates documentation for df.interpolate() methods #60227

### LINKED ISSUE #60227: DOC: Improve documentation df.interpolate() for methods   ‘time’, ‘index’ and ‘values’
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.interpolate.html

### Documentation problem

It is not possible to understand what exactly the method `interpolate` does from reading the documentation. See e.g. this SE post for more details

https://stackoverflow.com/questions/65511992/pandas-interpolation-type-when-method-index

### Suggested fix for documentation

Rewrite doctstring and documentation page for the method

### PR DESCRIPTION
**Pull request by anishkarki**
- [x] closes #60227 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### This PR improves the documentation for the `DataFrame.interpolate()` method and closes #60227

The focus is on providing more precise descriptions and examples for the following `method` options:

- **`method="index"`**: Emphasized that this method uses the actual numerical values of the index, which is particularly useful for non-sequential or time-based indices.
- **`method="time"`**: Expanded on the description to specify that this method interpolates based on the time intervals between observations, suitable for data with a time-based index.
- **`method="values"`**: Clarified that this method performs interpolation based on the actual numerical values within the DataFrame, treating them as equally spaced along the index.

### Changes Made

- Enhanced the docstring descriptions for `values`, `index`, and `time` methods in `DataFrame.interpolate()`.
- Added a practical example using a DataFrame with a non-sequential index to demonstrate the difference between `method="linear"` and `method="index"`.

### CODE REVIEW

**Comment by anishkarki** [comment]
@rhshadrach  Could you review this PR to improve documentation for df.interpolate() methods as part of issue #60227?”

File: pandas/core/generic.py [hunk 1]
```diff
@@ -7668,8 +7668,12 @@ def interpolate(
             * 'linear': Ignore the index and treat the values as equally
               spaced. This is the only method supported on MultiIndexes.
             * 'time': Works on daily and higher resolution data to interpolate
-              given length of interval.
-            * 'index', 'values': use the actual numerical values of the index.
+              given length of interval. This interpolates values based on
+              time interval between observations.
+            * 'index': The interpolation uses the actual numerical values
+              of the df's index to linearly calculate missing value.
```
Comment (rhshadrach) [inline pandas/core/generic.py:7674]:
```suggestion
            * 'index': The interpolation uses the numerical values
              of the DataFrame's index to linearly calculate missing values.
```

File: pandas/core/generic.py [hunk 2]
```diff
@@ -7668,8 +7668,12 @@ def interpolate(
...
+            * 'value': Interpolation based on the actual numerical values
+              in the DataFrame, treating them as equally spaced along the index.
```
Comment (rhshadrach) [inline pandas/core/generic.py:7676]:
```suggestion
            * 'values': Interpolation based on the numerical values
              in the DataFrame, treating them as equally spaced along the index.
```

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR! I see no difference between "actual numeric values" and "numeric values", so I think "actual" can be removed.

**Comment by anishkarki** [comment]
@rhshadrach  I've made the requested changes to the documentation. Could you please review it again? Thanks!

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @anishkarki

**Pull request merged**